randovania/games/super_metroid/patcher/super_duper_metroid_patcher.py

```diff
@@ -193,26 +256,6 @@ def patch_game(self, input_file: Optional[Path], output_file: Path, patch_data:
 
         temporary_output.write_bytes(vanilla_bytes)
 
-        item_list = [
-            _json_to_pickup_placement_data(pickup)
-            for pickup in patch_data["item_list"]
-        ]
-        starting_items = [
-            SuperDuperMetroid.ROM_Patcher.PickupPlacementData(
-                quantity_given=quantity,
-                pickup_index=-1,
-                item_name=name,
-            )
-            for name, quantity in patch_data["starting_items"]
-        ]
-
-        patches_to_apply = ["InstantG4", "MaxAmmoDisplay"]
-        SuperDuperMetroid.ROM_Patcher.patch_rom(
-            os.fspath(temporary_output),
-            item_list=item_list,
-            starting_items=starting_items,
-            skip_intro=True,
-            static_patches=patches_to_apply,
-        )
+        SuperDuperMetroid.ROM_Patcher.patch_rom_json(os.fspath(temporary_output), json.dumps(patch_data))
```

Comment:
Did you add an API where you can pass the rom `bytes` or `io.BytesIO` so we don't need a temporary file?

Reply:
No, should I do that?

Reply:
Preferable. Would let us not have the internal copy at all I think.